src/base/bittorrent/session.cpp

```diff
@@ -1984,6 +1984,25 @@ void Session::bottomTorrentsQueuePos(const QVector<TorrentID> &ids)
     saveTorrentsQueue();
 }
 
+void Session::handleTorrentNeedSaveResumeData(const TorrentImpl *torrent)
+{
+    if (m_needSaveResumeDataTorrents.empty())
+    {
+        QMetaObject::invokeMethod(this, [this]()
+        {
+            for (const TorrentID &torrentID : asConst(m_needSaveResumeDataTorrents))
+            {
```

Comment:
you're iterating on an empty array. (first-line tests for empty array)

Reply:
ohh you add then call in QueuedConnection. one shouldn't use Queued Connection like this.

Reply:
>one shouldn't use Queued Connection like this.

Why is that? Take the trouble to justify it. And at the same time, suggest a better way.

Reply:
As an alternative, I see only the following method:
1. Make TorrentImpl QObject
2. Declare `needSaveResumeData` signal
3. Declare some boolean to track whether `needSaveResumeData` is emitted but `saveResumeData()` isn't invoked yet.
4. Emit `needSaveResumeData` when some appropriate data is changed and set boolean flag to true
5. Session connects to `needSaveResumeData` signal using queued connection and call `saveResumeData()` in its handler
6. Boolean flag is cleared in `saveResumeData()`

Does that look preferable to you?
@Chocobo1?

Reply:
BTW, this alternative will look generally ugly if you apply it to another similar case, when it comes to reconfiguring the session when changing some properties.

Reply:
>Does that look preferable to you?

Subjectively, I would prefer it as is.